server/pxf-json/src/main/java/org/greenplum/pxf/plugins/json/JsonRecordReader.java

```diff
@@ -73,25 +89,27 @@ public JsonRecordReader(JobConf conf, FileSplit split) throws IOException {
 
         start = split.getStart();
         end = start + split.getLength();
-        final Path file = split.getPath();
+        file = split.getPath();
+        hosts = split.getLocations();
         compressionCodecs = new CompressionCodecFactory(conf);
         final CompressionCodec codec = compressionCodecs.getCodec(file);
 
         // openForWrite the file and seek to the start of the split
         FileSystem fs = file.getFileSystem(conf);
         FSDataInputStream fileIn = fs.open(split.getPath());
         if (codec != null) {
-            is = codec.createInputStream(fileIn);
             start = 0;
             end = Long.MAX_VALUE;
+            FileSplit codecSplit = new FileSplit(file, start, Long.MAX_VALUE, hosts);
```

Comment:
```suggestion
            FileSplit codecSplit = new FileSplit(file, start, Long.MAX_VALUE, split.getLocations());
```
`hosts` member is not used anywhere else, so might as well just inline it here